### ISSUE TITLE
gcp-gke-control-plane-auth-networks-updater should be more resilient

### ISSUE THREAD

**Issue by aptituz** [metro-digital/cf-github-actions#75]
We noticed that workflows that use gcp-gke-control-plane-auth-networks-updater are a bit brittle, because Google allows only one update operation at a time. So if multiple workflows are deploying to the same cluster the action can fail with an error message like this:

> 2023/03/21 10:45:01 ::error::Failed to update cluster: googleapi: Error 400: Operation operation-1679395497155-bea4d7ae is currently upgrading cluster companion-services-0h. Please wait and try again once it is done., failedPrecondition

This could be improved by adding retries with some delay, e.g. if the operation in progress error is thrown, the networks-updater shall wait some seconds and then retry up to n times. Ideally, the implementation would use a timeout with some jitter to avoid introducing a thundering herd problem.

This would potentially enlarge the time a pipeline runs but reduce manual interventions. At the moment, if a workflow fails due to this error it needs to be retried manually.

**Comment by aptituz** [comment]
@ippschi-mid and I recently worked on fixing this issue and it seems the issue might be trickier than initially thought.

Currently we have a version that is able to successfully update the cluster configuration most of the time, but it turns out that the pipeline may fail anyway. 

The current working hypothesis why this is the case is the following: every time one job updates the cluster configuration it takes a while until this cluster configuration is visible for subsequent get-requests. The update itself, however, needs the current configuration as the google ip always requires the complete list of IPs. Thus, while the add for an individual run might be successful, the add-operation of another job might get an outdated version of the current configuration and therefore overwrite this configuration again.

Basically, we never achieve a consistent state and thus the pipeline fails at the deployment stage, because it times out.

I'll try a few things more tomorrow, but I'm running out of ideas.

**Comment by aptituz** [comment]
We meanwhile found a version of the action that seems to do the trick. We are currently giving it a bit more testing and will provide a PR afterwards.
